docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md

```diff
@@ -0,0 +1,79 @@
+
+The files generated by this tool are a good reference of the files that can be involved in the definition and registration of a block.
+
+[![Open File Structure of a Block Diagram in excalidraw](https://developer.wordpress.org/files/2023/11/file-structure-block.png)](https://excalidraw.com/#json=YYpeR-kY1ZMhFKVZxGhMi,mVZewfwNAh_oL-7bj4gmdw "Open File Structure of a Block Diagram in excalidraw")
+
+### `<plugin-file>.php`
+
+A block is usually added to the block editor using a WordPress plugin. In the main PHP file of the plugin the block is usually registered on the server side.
+
+<div class="callout callout-info">
+For more on creating a WordPress plugin see <a href="https://developer.wordpress.org/plugins/plugin-basics/">Plugin Basics</a>, and <a href="https://developer.wordpress.org/plugins/plugin-basics/header-requirements/">Plugin Header requirements</a> for explanation and additional fields you can include in your plugin header.
+</div>
+
+### `package.json`
+
+[`package.json`](https://docs.npmjs.com/cli/v10/configuring-npm/package-json) is a configuration file for a NodeJs project. In this file you define the NPM dependencies of the block and the scripts used for local work.
+
+### `src` folder
+
+In a standard project you'll place your block files in the `src` folder. By default, the build process with `wp-scripts `will take files from this folder and will generate the bundled files in the `build` folder. 
+
+### `block.json`
+
+This files contains the metadata of the block, and it's required for its registration. Among other things it provides properties to define the files you need to be part of the definition of the block:
+- `editorScript` → `index.js` by default
+- `style` → `style-index.css` (output build for `style.scss`) by default
+- `editorStyle` → `index.css` (output build for `editor.scss`) by default
+- `render` → `render.php` by default
+- `viewScript` → `view.js` by default
+
+### `index.js`
+
+The `index.js` file (or any other file defined in the `editorScript` property of `block.json`) is the entry point file for javascript files and it usually registers the block in the client. In a standard structure it imports the `edit.js` and `save.js` files to get functions required in block registration.
+
+### `edit.js`
+
+The `edit.js` file usually returns the `edit` function used in block registration that defines the React component that will be loaded on the Editor when the block is loaded or inserted.
+
+### `save.js`
+
+The `save.js` file usually returns the `save` function used in block registration that defines the markup used to save the block in the DB.
+
+### `style.scss`
+
+The `style.scss` file contains the styles of the block that will be loaded in both the editor and the frontend. In the build process this file is converted into `style-index.css` which is usually defined at `style` property in `block.json`
```

Comment:
Will it be clear enough that this is a scss / sass file, or do we need to add something more about the file format and the processing?

Reply:
I have added an info box providing some more context about these `.scss` files 👍